src/sources/util/unix_datagram.rs

```diff
@@ -26,16 +27,21 @@ use crate::{
 /// syslog source).
 pub fn build_unix_datagram_source(
     listen_path: PathBuf,
+    socket_file_mode: Option<u32>,
     max_length: usize,
     decoder: Decoder,
     handle_events: impl Fn(&mut [Event], Option<Bytes>) + Clone + Send + Sync + 'static,
     shutdown: ShutdownSignal,
     out: SourceSender,
-) -> Source {
-    Box::pin(async move {
-        let socket = UnixDatagram::bind(&listen_path).expect("Failed to bind to datagram socket");
-        info!(message = "Listening.", path = ?listen_path, r#type = "unix_datagram");
+) -> crate::Result<Source> {
+    let socket = UnixDatagram::bind(&listen_path).expect("Failed to bind to datagram socket");
+    info!(message = "Listening.", path = ?listen_path, r#type = "unix_datagram");
 
+    if let Err(e) = change_socket_permissions(&listen_path, socket_file_mode) {
+        return Err(e);
+    }
```

Comment:
This has a builtin form:
```suggestion
    change_socket_permissions(&listen_path, socket_file_mode)?;
```